gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/IndexGigawordWithElasticSearch.java

```diff
@@ -1,52 +1,151 @@
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.io.File;
-import java.io.IOException;
-import java.util.Map;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.UnmodifiableIterator;
+
+import edu.isi.nlp.parameters.Parameters;
+import edu.isi.vista.gigawordIndexer.GigawordFileProcessor.Article;
 
 public class IndexGigawordWithElasticSearch {
-    private static final ImmutableMap<String, String> sampleDocs = ImmutableMap.of(
-            "doc1", "Hello world",
-            "doc2", "la la la",
-            "doc3", "The music-room in the Governor’s House at Port Mahon, a tall, handsome," +
-                    " pillared octagon, was filled with the triumphant first movement of Locatelli’s C major quartet."
-    );
-
-    public static void main(String[] argv) throws IOException {
-        final Parameters parameters;
-
-        if (argv.length > 0) {
-            parameters = Parameters.loadSerifStyle(new File(argv[0]));
-        } else {
-            parameters = Parameters.fromMap(ImmutableMap.of());
-        }
-
-        try (RestHighLevelClient client = new RestHighLevelClient(
-                RestClient.builder(
-                        new HttpHost(parameters.getOptionalString("primaryHostName").or("localhost"),
-                                parameters.getOptionalPositiveInteger("primaryPort").or(9200),
-                                "http"),
-                        new HttpHost(parameters.getOptionalString("secondaryHostName").or("localhost"),
-                                parameters.getOptionalPositiveInteger("secondaryPort").or(9201),
-                                "http")
-                ))) {
-            for (Map.Entry<String, String> e : sampleDocs.entrySet()) {
-                final String docId = e.getKey();
-                final String docContent = e.getValue();
-
-                // you can index documents one-by-one
-                client.index(new IndexRequest("documents", "doc", docId)
-                        .source("text", docContent));
-            }
-
-            // TODO: in practice, you will want to use BulkRequest for speed
-        }
+
+  private static final Logger log = LoggerFactory.getLogger(IndexGigawordWithElasticSearch.class);
+
+  private static final String DEFAULT_INDEX_NAME = "Gigaword";
+
+  private static final String DEFAULT_HOST = "localhost";
+
+  private static final int DEFAULT_PORT_PRI = 9200;
+
+  private static final int DEFAULT_PORT_SEC = 9201;
+
+  private static final String PARAM_FILEPATH = "filePath";
+
+  private static final String PARAM_INDEX_NAME = "indexName";
+
+  private static final String PARAM_HOSTNAME_PRI = "primaryHostName";
+
+  private static final String PARAM_HOSTNAME_SEC = "secondaryHostName";
+
+  private static final String PARAM_PORT_PRI = "primaryPort";
```

Comment:
Prefer to spell out `PRIMARY` and `SECONDARY`.  Bytes are cheap, confusion is expensive :-)